### PR TITLE
UILivePreview component snippet wrapper

### PR DESCRIPTION
**Pull request by adids1221**
## Description
Docs playground (`live-code`) device wrapper to simulate `Mobile` device.
In the `buildDocsCommon` script wrapping the component snippet with the `UILivePreview` that render `react-live` preview using iframe.
`livePreviewSource` set in the `docusaurus.config` depend on `NODE_ENV`.

## Changelog
Docs playground (`live-code`) device wrapper to simulate `Mobile` device.

## Additional info
MADS-4507

### CODE REVIEW

File: src/components/sortableGridList/sortableGridList.api.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
     "",
     "  return (",
     "    <div>",
-    "      <View style={{width: 800}}>",
+    "      <View style={{width: 300}}>",
```
Comment (M-i-k-e-l) [inline src/components/sortableGridList/sortableGridList.api.json:62]: We need to remove this and the `containerWidth={800}`

File: docuilib/src/pages/livePreview.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React, {useEffect, useState} from 'react';
+import {LiveProvider, LivePreview} from 'react-live';
+import ReactLiveScope from '../theme/ReactLiveScope';
+
+const messageType = 'LIVE_PREVIEW_CODE_UPDATE_MESSAGE';
```
Comment (M-i-k-e-l) [inline docuilib/src/pages/livePreview.tsx:5]: Can we unify these to a single place?

File: docuilib/src/components/UILivePreview.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  const iframeRef = useRef(null);
+  const iframeSource = siteConfig?.customFields?.livePreviewSource as string;
+
+  useEffect(() => {
+    if (iframeRef.current && iframeLoaded) {
+      setTimeout(() => {
+        sendMessageToIframe(code);
+      }, 100);
+    }
+  }, [iframeRef, iframeLoaded, code]);
```
Comment (M-i-k-e-l) [inline docuilib/src/components/UILivePreview.tsx:23]: Try to remove `iframeRef`, AFAIK `ref`s should not be added as a `dep`

File: docs/getting-started/usage.md [hunk 4]
```diff
@@ -5,32 +5,32 @@ title: 'Usage'
...
+<SegmentedControl segments={[{label: 'Register'}, {label: 'Login'}]} />
+<View marginT-s5>
+<TextField preset="outline" placeholder="username" />
+<TextField preset="outline" placeholder="password" secureTextEntry grey10 />
+</View>
+<View row marginT-s5 centerV>
+<Button link text70 orange30 label="Sign Up" marginR-s5 />
+<Button text70 white background-orange30 label="Login" />
+</View>
+</View>
```
Comment (M-i-k-e-l) [inline docs/getting-started/usage.md:32]: Please re-indent

File: docuilib/src/components/UILivePreview.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    <View row gap-s2 style={styles.liveCodeWrapper}>
+      <LiveProvider code={code} scope={ReactLiveScope}>
+        <View flex style={styles.editorWrapper}>
+          <LiveEditor
+            className="font-mono"
+            onChange={setCode}
+            style={{
+              overflowY: 'scroll',
+              scrollbarWidth: 'none' // For Firefox
+            }}
```
Comment (M-i-k-e-l) [inline docuilib/src/components/UILivePreview.tsx:40]: Lets move to the `styles` (you already have it here)

File: docuilib/src/pages/livePreview.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    window.addEventListener('message', (e: MessageEvent) => {
+      if (e.data.type === messageType) {
+        setCode(e.data.code);
+      }
+    });
+  }, []);
+
+  return (
+    <LiveProvider code={code} scope={ReactLiveScope}>
+      <LivePreview style={{overflow: 'hidden'}}/>
```
Comment (M-i-k-e-l) [inline docuilib/src/pages/livePreview.tsx:20]: Another style :pray:

File: docuilib/src/components/UILivePreview.tsx (hunk 5 above)
Comment (adids1221) [reply docuilib/src/components/UILivePreview.tsx:40]: moved into `useMemo`

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**